### PR TITLE
Add nhwc support for QLinearAveragePool operator

### PR DESCRIPTION
**Pull request by zhanghuanrong**
* Add channels_last to QLinearAveragePool  operator definition
* Implement nhwc format QLinearAveragePool operator logic
* Unit test for QLinearAveragePool operator for nhwc format data
* Unit test for nhwc transformer for QLinearAveragePool

### CODE REVIEW

File: onnxruntime/core/graph/contrib_ops/nhwc_schema_defs.cc [hunk 1]
```diff
@@ -263,6 +263,156 @@ equal to the spatial dimension of input tensor. Input is of type uint8_t or int8
           ++image_dim_index;
         }
       });
+
+  const char* QLinearAveragePoolDoc_ver1 = R"DOC(
```
Comment (yufenglee) [inline onnxruntime/core/graph/contrib_ops/nhwc_schema_defs.cc:267]: nit: static

Comment (zhanghuanrong) [reply onnxruntime/core/graph/contrib_ops/nhwc_schema_defs.cc:267]: nice. Fixed it.

File: onnxruntime/contrib_ops/cpu/qlinear_pool.cc [hunk 2]
```diff
@@ -82,15 +93,76 @@ struct QLinearPool1DTask final {
...
+    for (int64_t remains = end - begin; remains > 0; offset = 0, batch++) {
+      if (offset + remains <= y_image_size) {
+        operator()(std::ptrdiff_t(batch), std::ptrdiff_t(offset), std::ptrdiff_t(offset + remains));
+        remains = 0;
+      } else {
+        operator()(std::ptrdiff_t(batch), std::ptrdiff_t(offset), std::ptrdiff_t(y_image_size));
+        remains -= (y_image_size - offset);
+      }
+    }
+  }
```
Comment (yufenglee) [inline onnxruntime/contrib_ops/cpu/qlinear_pool.cc:130]: nit:this logic to calculate image to process can be shared

**Review by yufenglee — APPROVED** [review]
:shipit:

**Pull request merged**